sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -0,0 +1,267 @@
+    classOf[JArrayList[String]])
+  private lazy val loadPartitionMethod = findMethod(classOf[Hive], "loadPartition", classOf[Path],
+    classOf[String], classOf[JMap[String, String]], JBoolean.TYPE, JBoolean.TYPE,
+    JBoolean.TYPE, JBoolean.TYPE)
+  private lazy val loadTableMethod = findMethod(classOf[Hive], "loadTable", classOf[Path],
+    classOf[String], JBoolean.TYPE, JBoolean.TYPE)
+  private lazy val loadDynamicPartitionsMethod = findMethod(classOf[Hive], "loadDynamicPartitions",
+    classOf[Path], classOf[String], classOf[JMap[String, String]], JBoolean.TYPE,
+    JInteger.TYPE, JBoolean.TYPE, JBoolean.TYPE)
+
+  override def setCurrentSessionState(state: SessionState): Unit = startMethod.invoke(null, state)
+
+  override def getDataLocation(table: Table): Option[String] =
+    Option(getDataLocationMethod.invoke(table)).map(_.toString())
+
+  override def setDataLocation(table: Table, loc: String): Unit =
+    setDataLocationMethod.invoke(table, new URI(loc))
+
+  override def getAllPartitions(hive: Hive, table: Table): Seq[Partition] =
+    getAllPartitionsMethod.invoke(hive, table).asInstanceOf[JSet[Partition]].toSeq
+
+  override def getCommandProcessor(token: String, conf: HiveConf): CommandProcessor =
+    getCommandProcessorMethod.invoke(null, token, conf).asInstanceOf[CommandProcessor]
+
+  override def getDriverResults(driver: Driver): Seq[String] = {
+    val res = new JArrayList[String]()
+    getDriverResultsMethod.invoke(driver, res)
+    res.toSeq
+  }
+
+  override def loadPartition(
+      hive: Hive,
+      loadPath: Path,
+      tableName: String,
+      partSpec: JMap[String, String],
+      replace: Boolean,
+      holdDDLTime: Boolean,
+      inheritTableSpecs: Boolean,
+      isSkewedStoreAsSubdir: Boolean): Unit = {
+    loadPartitionMethod.invoke(hive, loadPath, tableName, partSpec, replace: JBoolean,
+      holdDDLTime: JBoolean, inheritTableSpecs: JBoolean, isSkewedStoreAsSubdir: JBoolean)
+  }
+
+  override def loadTable(
+      hive: Hive,
+      loadPath: Path,
+      tableName: String,
+      replace: Boolean,
+      holdDDLTime: Boolean): Unit = {
+    loadTableMethod.invoke(hive, loadPath, tableName, replace: JBoolean, holdDDLTime: JBoolean)
+  }
+
+  override def loadDynamicPartitions(
+      hive: Hive,
+      loadPath: Path,
+      tableName: String,
+      partSpec: JMap[String, String],
+      replace: Boolean,
+      numDP: Int,
+      holdDDLTime: Boolean,
+      listBucketingEnabled: Boolean): Unit = {
+    loadDynamicPartitionsMethod.invoke(hive, loadPath, tableName, partSpec, replace: JBoolean,
+      numDP: JInteger, holdDDLTime: JBoolean, listBucketingEnabled: JBoolean)
+  }
+
+}
+
+private[client] class Shim_v0_13 extends Shim_v0_12 {
+
+  private lazy val setCurrentSessionStateMethod =
+    findStaticMethod(classOf[SessionState], "setCurrentSessionState", classOf[SessionState])
+  private lazy val setDataLocationMethod = findMethod(classOf[Table], "setDataLocation",
+    classOf[Path])
```

Comment:
This is a nit, but the indenting style here is off and inconsistent with the above line (which would be my preference).  In SQL we usually use the following rules (which attempt to avoid having some arguments look 'special' due to indenting):
 - Break at the highest syntactic level (i.e. at the `=` not in the middle of a function argument list).
 - If you have to wrap arguments, wrap all arguments.